distributed/diagnostics/bokeh/status/main.py

```diff
@@ -35,4 +39,14 @@ def task_update():
         task_table_update(task_source, msg)
 doc.add_periodic_callback(task_update, messages['tasks']['interval'])
 
-doc.add_root(vplot(worker_table, task_table))
+resource_source, resource_plot = resource_profile_plot()
+@gen.coroutine
+def resource_update():
+    with log_errors():
+        yield messages['workers']['condition'].wait()
+        worker_buffer = messages['workers']['deque']
+        times_buffer = messages['workers']['times']
+        resource_profile_update(resource_source, worker_buffer, times_buffer)
```

Comment:
Do we want the entire history?  We could either:

1.  Take only the last n messages (probably also much cheaper this way)
2.  Use a DataRange1d follow/follow_interval and xwheel_zoom